### PR TITLE
Deactivation of password grant

### PR DESCRIPTION
**Pull request by sftschaerff**
- added warn message, when one tries to authenticate with user and userPassword
- removed unnecessary code
- added section "IMPORTANT NOTE" to README file

### CODE REVIEW

File: lib/auth.js [hunk 1]
```diff
@@ -158,7 +158,12 @@ function obtainToken(accountManagerHostOverride, basicAuthUser, basicAuthPasswor
  */
 function auth(client, clientSecret, user, userPassword, autoRenew, accountManager) {
     // determine oauth flow to use, by default it is resource owner password credentials
-    var flow = { grant : 'password', response_type : 'code' };
+    var flow = { grant : 'client_credentials', response_type : 'token' };
+
+    if ( user || userPassword ) {
```
Comment (challmannsf) [inline lib/auth.js:163]: @sftschaerff  

I'm not 100% sure if we switch the clientId/secret combination which was using *password* grant before, does still work for *client_credentials* without changing the related account in the AM.
Did you test that?

Maybe we can just add some sort of deprecation message and leave the flow as it is currently?
So in the end of the file we could place a warning and later on remove the related code?
````
if (flow.grant === 'password') {
  console.warn('some deprecation msg')
}
```

Comment (sftschaerff) [reply lib/auth.js:163]: As of Gerrit, the client (with the related clientID) should have the role "Account Administrator" and then this should work the same like the old _password_ grant. I have added this requirement to the README section. I haven't tested it.

**Comment by sholsinger** [comment]
@challmannsf I know the `password` grant isn't documented anywhere... but I just began using it to automate the startup/shutdown of sandboxes without MFA. (Because I reverse engineered sfcc-ci's usage of it.) Does the `client_credentials` grant work the same way?

**Comment by gokaygurcan** [comment]
We're having a similar situation. I am one of the three account administrators in our organization, and _have to_ deal with this non-sense MFA every time I log in through web browser or via CLI. That conflicts with the very purpose of this tool: automation. 

To overcome this problem, we created a dummy user, gave it some limited access, and using it to log in without thinking about MFA. But if SF decides to enforce it on all users at some point in the future, that'll be a real challenge..

**Comment by Trimud** [comment]
I share the same concerns. We have built React app with user roles, assignments, scheduling etc. and I don't see how this will work if you enforce 2FA for everything.

**Comment by jorgehernandezSF** [comment]
Adding @tobiaslohr for awareness of this discussion.

**Comment by jorgehernandezSF** [comment]
Hello all,
There is a related discussion in the partner community around this:
https://partners.salesforce.com/_ui/core/chatter/groups/GroupProfilePage?g=0F93A000000DQ6f&fId=0D54V000054q0L4&s1oid=00D300000000iTz&s1nid=0DB3000000007Uh&emkind=chatterGroupDigest&s1uid=0053A00000DgWfy&emtm=1607416240140&fromEmail=1&s1ext=0

John Zhao (SF) suggestion is:  As to your question, we are enhancing the sandbox API to accept Oauth client_credential grants (ETA Jan 2021). In the meantime, a possible workaround is: set up a service (or automation) user in Account Manager who only has the Sandbox_API_User role and toggle off MFA for that user in your Account Manager org. I hope that helps.

**Comment by tobiaslohr** [comment]
@sftschaerff @sholsinger @gokaygurcan @Trimud @challmannsf @jorgehernandezSF To make things clear: This PR will not be merged! The Oauth resource_owner_password_grant is still supported by Commerce Cloud Account Manager and is also needed to automate usage of the Sandbox API. Reason being is the need for service user, as today the Sandbox API does not support a client_credentials grant (which only requires an API client, without a user).

The enhancement for the Sandbox API is on the product team's backlog and will be worked on with priority. Until then the Account Manager will still support that grant type hence our CLI will also implement and offer this flow.

**Comment by tobiaslohr** [comment]
We will close this PR without merging the changes (see my earlier comment).

**Pull request closed without merging**